coops-ui/src/components/AnnualReport/Directors.vue

```diff
@@ -917,7 +911,7 @@ export default class Directors extends Mixins(DateMixin, ExternalMixin, EntityFi
         lastName: this.director.officer.lastName
       },
       deliveryAddress: { ...this.inProgressAddress },
-      mailingAddress: { ...this.inProgressMailAddress },
+      ...(this.entityFilter(EntityTypes.BCorp) && { mailingAddress: { ...this.inProgressMailAddress } }),
       appointmentDate: this.asOfDate, // when implemented: this.director.appointmentDate,
```

Comment:
@severinbeauvais 
I updated the condition to be specific to BCORPs as it is sort of arbitrary until we have more information and if you prefer it his, no problem at all! 

I disagree that this method is too complicated however. Using rest and spread operators are quite common now and it seems quite useful in this case. A ternary operator will not actually omit the mailingAddress property whereas wrapping the entire `newDirector` obj into an if else statement is writing duplicate code in my opinion. Is this perfect or the best way, i'm not sure,  so I'm certainly open to ideas though of what you think would be simpler than this.

Reply:
I had a quick poll here and all agreed that this may work but it's difficult to understand. (Sorry.) The good news is we are all learning something from your fancy code :)

An option is to conditionally add this property to the newDirector object. Thoughts?

Reply:
I had a quick poll here and all agreed that this may work but it's difficult to understand. (Sorry.) The good news is we are all learning something from your fancy code :)

An option is to conditionally add this property to the newDirector object. Thoughts?

Reply:
well of course it works 😆  Yes we can assign the property conditionally after creating it, if that is what the team prefers :)

Reply:
I didn't check with the team but I prefer this -- thanks!